webapp/updates.py

```diff
@@ -49,12 +50,13 @@ def _process_repositories(self, data, response):
 
     def _process_input_packages(self, data, response):
         """Parse input NEVRAs and filter out unknown (or without updates) package names."""
-        packages_to_process = data.get('package_list', None)
+        latest_only = data.get("latest_only", False)
+        packages_to_process = filter_package_list(data.get('package_list', None), latest_only)
         filtered_packages_to_process = {}
         if packages_to_process is not None:
             for pkg in packages_to_process:
                 response['update_list'][pkg] = {}
-                name, epoch, ver, rel, arch = split_packagename(pkg)
+                name, epoch, ver, rel, arch = parse_rpm_name(pkg, default_epoch='0')
```

Comment:
Assuming `latest_only` is True, `parse_rpm_name` is called twice for every pkg, first in `filter_package_list` function, then here. It'd be good if the previously parsed result was re-used.

Reply:
I'm aware of it.
Unnecessary parsing of 1k package names takes extra 2.1ms.
Returning already parsed list would make the code complicated and required joining components back (L#56) which takes 1.9ms for 1k packages.
Given the above I've decided to keep more readable code.

Reply:
ok, fair enough, I've also haven't measured any serious performance hit